services/catarse/app/actions/portal/users/authenticate_by_token.rb

```diff
@@ -0,0 +1,20 @@
+# frozen_string_literal: true
+
+module Portal
+  module Users
+    class AuthenticateByToken < Actor
+      input :authorization_header, type: String, allow_nil: true
+
+      output :user, type: [NilClass, User]
+
+      def call
+        auth_token = authorization_header.to_s.split.last
+        fail!(error: 'Missing auth token') if auth_token.blank?
+        payload = Portal::AuthToken.decode(auth_token: auth_token)
+        self.user = User.find(payload.first['user_id'])
+      rescue ActiveRecord::RecordNotFound, JWT::VerificationError, JWT::ExpiredSignature
```

Comment:
Aqui n seria melhor colocar um erro padrão tbm, para evitar que outro tipo de erro tbm enviasse a mensagem de fail.

Reply:
@gabrielras nesse caso não. Nesse caso eu quero que esses erros específicos (token expirado, token inválido e usuário do token não encontrado) sejam tratados  e que a ação retorne que o auth token é inválido. Se acontecer outro erro além desse, por exemplo, por exemplo: `undefined method split for nil`, que isso seja reportado como internal server error, pois não é o auth token que está inválido, e sim um erro de implementação nosso que precisamos corrigir e tratar.